### PR TITLE
Add tacotron2 to TTS2

### PR DESCRIPTION
**Pull request by jctian98**
## What?

[WIP] Add tacotron2 recipes to TTS2 toolkit

@ftshijt

### CODE REVIEW

File: espnet2/bin/tts2_inference.py [hunk 1]
```diff
@@ -141,6 +142,8 @@ def __call__(
         batch = dict(text=text)
         if speech is not None:
             batch.update(speech=speech)
+        if discrete_speech is not None:
```
Comment (ftshijt) [inline espnet2/bin/tts2_inference.py:145]: We might consider beamsearch for the AR cases?

File: espnet2/tts2/tacotron2/loss.py [hunk 2]
```diff
@@ -0,0 +1,133 @@
+import torch
```
[the remark this answers is not in the document]
Comment (jctian98) [reply espnet2/tts2/tacotron2/loss.py:1]: Yes. I'll have a try. It's still under testing.
